src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -295,5 +295,47 @@ public static void CopyTo<T>(this T[] array, Span<T> destination)
         {
             new ReadOnlySpan<T>(array).CopyTo(destination);
         }
+
+        /// <summary>
+        /// Determines whether two sequences overlap.
+        /// </summary>
+        public static unsafe bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second)
```

Comment:
Is this overload needed? The ` public static unsafe bool Overlaps<T>(this ReadOnlySpan<T> first, ReadOnlySpan<T> second)` method should be enough, since Span implicitly converts to ReadOnlySpan, or is there a specific reason for this?